overlord/devicestate/devicemgr.go

```diff
@@ -991,6 +992,55 @@ func (m *DeviceManager) ensureSeeded() error {
 	return nil
 }
 
+var processAutoImportAssertionsImp = processAutoImportAssertions
+
+// ensureAutoImportAssertions makes sure that auto import assertions
+// get processed
+func (m *DeviceManager) ensureAutoImportAssertions() {
+	// on classic do nothing
+	if release.OnClassic {
+		return
+	}
+
+	// ignore while we are in preseeding
+	if m.preseed {
+		return
+	}
+
+	// check if we have processed auto import as already
+	var asState string
+	err := m.state.Get("auto-as-imported", &asState)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		logger.Noticef("failed to get auto-import assert state")
+		return
+	}
+
+	if asState == "done" || asState == "pending" {
+		return
+	}
+
+	cached := m.state.Cached(loadedDeviceSeedKey{})
+	if cached != nil {
+		loaded := cached.(*loadedDeviceSeed)
```

Comment:
the cache management for this is complicated, at some point after seeding we clear the cache. I need to think because I don't remember all the details also I kind of wanted to simplify it a bit as well, because there is some double caching going on now but it requires turning populateStateFromSeed into a DeviceManager method, maybe I should bite the bullet and do that.

Reply:
I'm piggy backing in purpose here, as I want to avoid loading seed.
I can't find where we load see on preseeded cases.
From my tests however this was able to pick seed from cache every time.
But I am all ears how to use cache in better way :D